cypress/e2e/voting_spec.cy.js

```diff
@@ -1,45 +1,92 @@
         "poster_path": "https://image.tmdb.org/t/p/original//vfJFJPepRKapMd5G2ro7klIRysq.jpg",
         "title": "Your Name.",
-        "vote_count": 11242
+        "vote_count": 11243
       }
-    }).as('posterVote')
-  })
+    }).as('posterVote');
 
-  it('increases the vote count when clicking the upvote button', () => {
-    cy.get('.upvote-button').should('exist')
+    cy.get(':nth-child(5) > .vote-bar > .vote-count').should('have.text', '11242')
+    cy.get(':nth-child(5) > .vote-bar > .upvote-button').click()
+    cy.wait('@posterVote')
+    cy.get('@posterVote').its('response.statusCode').should('eq', 200)
+    cy.get(':nth-child(5) > .vote-bar > .vote-count').invoke('text').then(text => {
+      const newVoteCount = (parseInt(text) + 1).toString()
+      cy.get(':nth-child(5) > .vote-bar > .vote-count').should('have.text', '11243')
+    });
+  });
 
-    cy.get(':nth-child(55) > .vote-bar > .vote-count').should('have.text', '11242')
+  it('decreases the vote count when clicking the downvote button', () => {
+    cy.intercept('PATCH', 'https://rancid-tomatillos-api-cc6f59111a05.herokuapp.com/api/v1/movies/372058', {
+      statusCode: 200,
+      body: {
+        "id": 372058,
+        "poster_path": "https://image.tmdb.org/t/p/original//vfJFJPepRKapMd5G2ro7klIRysq.jpg",
+        "title": "Your Name.",
+        "vote_count": 11241
+      }
+    }).as('posterVote')
 
-    cy.get(':nth-child(55) > .vote-bar > .upvote-button').click()
+    cy.get(':nth-child(5) > .vote-bar > .vote-count').should('have.text', '11242')
+    cy.get(':nth-child(5) > .vote-bar > .downvote-button').click()
     cy.wait('@posterVote')
     cy.get('@posterVote').its('response.statusCode').should('eq', 200)
+    cy.get(':nth-child(5) > .vote-bar > .vote-count').invoke('text').then(text => {
+      const newVoteCount = (parseInt(text) - 1).toString()
+      cy.get(':nth-child(5) > .vote-bar > .vote-count').should('have.text', '11241')
+    })
+  });
+});
 
-    cy.get(':nth-child(55) > .vote-bar > .vote-count').invoke('text').then(text => {
-      const newVoteCount = parseInt(text) + 1
-      cy.get(':nth-child(55) > .vote-bar > .vote-count').should('have.text', newVoteCount.toString())
+
+describe('Voting Sad Paths', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000')
+    cy.intercept('GET', 'https://rancid-tomatillos-api-cc6f59111a05.herokuapp.com/api/v1/movies', {
+      statusCode: 200,
+      fixture: 'movie_posters'
+    });
+    cy.intercept('PATCH', 'https://rancid-tomatillos-api-cc6f59111a05.herokuapp.com/api/v1/movies/372058', {
+      statusCode: 400,
+      body: {
+        "message": "Oops something went wrong... Try again later"
+      }
+    }).as('failedVote')
+  });
+
+  it('shows and alert when an error occures on the upvote', () => {
+    cy.get(':nth-child(5) > .vote-bar > .vote-count').should('have.text', '11242')
+    cy.get(':nth-child(5) > .vote-bar > .upvote-button').click()
+    cy.wait('@failedVote')
```

Comment:
Glad we got the opportunity to implement aliases in our cypress tests. A possible refactor would be to utilze these more, because I know i did not make use of these so far